web-components/src/utils/locationStandard.ts

```diff
@@ -49,3 +48,45 @@ export const getISOString = async (
   // If country-only, mapbox returns lowercase ('us'), so need toUppercase here for ledger
   return Promise.resolve(placeCode?.toUpperCase());
 };
+
+// Util function to prepare jurisdiction ISO code
+// based on package iso-3166-2
+
+type LocationType = {
+  country: string;
+  stateProvince?: string;
+  postalCode?: string;
+};
+
+export const getJurisdictionIsoCode = ({
+  country, // US
+  stateProvince, // US-CO
+  postalCode,
+}: LocationType): string => {
+  let jurisdiction = '';
+
+  // check iso country
+  const validCountry = iso3166.country(country);
+  if (!validCountry) return jurisdiction;
+
+  jurisdiction = country;
+
+  // check subdivision exists
+  if (!stateProvince) return jurisdiction;
+
+  // check iso subdivision
+  const validSubdivision = iso3166.subdivision(stateProvince);
+  if (!validSubdivision) return jurisdiction;
+
+  // TODO - text fields allow whitespace strings..
+  const _postalCode = postalCode?.trim();
+
+  // check postal code exists
+  if (_postalCode) {
+    jurisdiction = `${stateProvince} ${_postalCode}`;
+  } else {
+    jurisdiction = stateProvince as string;
+  }
+
+  return jurisdiction;
+};
```

Comment:
variable reassignment here makes the function a bit hard to follow, becuase you have to understand the entire scope to know what the return value will be.  If you check the same conditions and return once they aren't met, it’s easier to follow the logic:

```suggestion
export const getJurisdictionIsoCode = ({
  country, // US
  stateProvince, // US-CO
  postalCode,
}: LocationType): string => {
  if (!iso3166.country(country)) throw new Error(`Invalid country: ${country}`);
  if (!stateProvince || iso3166.subdivision(stateProvince)) return country;
  if (!postalCode) return stateProvince;
  return `${stateProvince} ${postalCode.trim()}`;
};
```
Also not sure we want this to return an empty string, so replaced with an `Error`

Reply:
That seems very good to me. Thanks for the proposed improvements @haveanicedavid !